### PR TITLE
Add class to manually validate rest data transfer

### PR DESCRIPTION
**Pull request by pfeuffer**
## Proposed changes

It may be necessary to validate objects annotated wit javax validation
tags, that could not be validated using the internal resteasy validation
mechanism, eg. objects created manually. This new class makes this
possible by simply calling validate(dto).

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] PR is well described
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [x] New code is covered with unit tests
- [x] CHANGELOG.md updated
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package sonia.scm.web.api;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import java.util.Set;
+
+public class DtoValidator {
```
Comment (sdorra) [inline scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java:34]: Is this only usable for dtos? I think the class could be final with a private constructor and static methods.

File: scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  void validate(Object configuration) {
```
Comment (sdorra) [inline scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java:35]: I think the method could be static and public

File: scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java (hunk 1 above)
Comment (sdorra) [reply scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java:34]: Does a unit test make sense?

Comment (pfeuffer) [reply scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java]: You're right :+1:

File: scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java (hunk 2 above)
Comment (pfeuffer) [reply scm-core/src/main/java/sonia/scm/web/api/DtoValidator.java]: Jepp....

File: scm-core/pom.xml [hunk 3]
```diff
@@ -194,6 +194,19 @@
...
+      <artifactId>javax.el-api</artifactId>
+      <version>3.0.0</version>
+      <scope>test</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.glassfish</groupId>
+      <artifactId>javax.el</artifactId>
+      <version>3.0.1-b11</version>
+    </dependency>
```
Comment (sdorra) [inline scm-core/pom.xml:208]: Is this dependency requried? Or only for testing?

**Pull request merged**